sdk/identity/identity/src/credentials/environmentCredential.ts

```diff
@@ -109,8 +133,13 @@ export class EnvironmentCredential implements TokenCredential {
       }
     }
 
-    span.setStatus({ code: CanonicalCode.UNAUTHENTICATED });
-    span.end();
-    return Promise.resolve(null);
+    // If by this point we don't have a credential, throw an exception so that
+    // the user knows the credential was not configured appropriately
+    throw new AuthenticationError(400, {
```

Comment:
should we validate that the array isn't somehow empty before throwing this?

Reply:
I don't think it'd give us any extra benefit since it would imply that the user had assigned every possible environment variable, meaning that `ClientSecretCredential` would already be assigned to `_credential`, skipping this throw entirely.

Reply:
Side note: I think maybe I shouldn't have deleted the `span.setStatus` and `span.end` before throwing.  Should I put them back?

Reply:
Yes please

Reply:
My one concern about this error message is that it doesn't give a clear picture of what sets are valid, it just tells you all the variables which could possibly be set. Here's the exception message we're using in other languages currently.

> "Environment variables not fully configured. To authenticate a service principal AZURE_TENANT_ID, AZURE_CLIENT_ID, and iether AZURE_CLIENT_SECRET or AZURE_CLIENT_CERTIFICATE_PATH must be set. Or to authenticate a user AZURE_TENANT_ID, AZURE_USERNAME and AZURE_PASSWORD must be set. Currently set variables [ ... "

Reply:
Thanks @schaabs, I'll expand on it!